atomix/cluster/src/test/java/io/atomix/raft/roles/LeaderRoleTest.java

```diff
@@ -344,6 +347,32 @@ public void onWriteError(final Throwable error) {
     verify(leaderRole.raft, timeout(2000).atLeast(1)).transition(Role.FOLLOWER);
   }
 
+  @Test
+  public void shouldSetCommitIndexOnSingleNode() throws InterruptedException {
```

Comment:
Is this test really needed? This is too low level test and difficult to maintain. I would recommend to keep these kind of tests only if there are no other ways to test it. For this case general raft test are enough to see if events are committed, isn't it?

Reply:
I agree that it's likely to break if we change stuff internally but I wanted to test that set commit (and flush) are called at the correct time which I can't do at a higher level. But I can remove it if you want

Reply:
Why is it important that setCommitIndex is called at correct time? Isn't it enough to verify that events are eventually committed?

Reply:
Because we set the last written index there and we have to do that before acknowledging the append. This doesn't directly test that the last written index is written and flushed but it tells us that the commit index would be called. It's not ideal but it seemed better than not having it. I don't mind deleting it but having a higher level test also won't tell us much

Reply:
I would recommend to remove this test as I don't see much value in it. Like you said it is only testing if the commit index is set and it assumes if commitIndex is set it would be flushed and last index is updated. So there is no point in testing it.

Reply:
I understand, I'll remove it then